packages/next/src/build/webpack-config.ts

```diff
@@ -1188,7 +1188,7 @@ export default async function getBaseWebpackConfig(
           chunks: isRspack
             ? // using a function here causes noticable slowdown
               // in rspack
-              /(?!polyfills|main|pages\/_app)/
+              /^(?!(polyfills|main|pages\/_app)$).*$/
```

Comment:
This seems bizarre? The regex needs to match the entire string?

Also, this regex looks equivalent to the function below, can we share the webpack and rspack codepaths?